includes/wf_crm_webform_preprocess.inc

```diff
@@ -529,7 +529,21 @@ class wf_crm_webform_preprocess extends wf_crm_webform_base {
           // If the user has already entered a value for this field, don't change it
           if (isset($this->info[$ent][$c][$table][$n][$name])
             && !(isset($component['cid']) && isset($submitted[$component['cid']]))) {
-            $val = $this->info[$ent][$c][$table][$n][$name];
+            // pick phone number and email according to selecte location type
+            if (($table == 'phone' && $name == 'phone')
+              || ($table == 'email' && $name == 'email')) {
+              $location_type_id = $this->node->webform_civicrm['data']['contact'][$c][$name][$n]['location_type_id'];
```

Comment:
you can't assume this variable is set - location_type_id is optional (could be "user select")